src/endpoints/renewal/get_non_subscribed_domains.rs

```diff
@@ -0,0 +1,95 @@
+use crate::{
+    models::AppState,
+    utils::{get_error, to_hex},
+};
+use axum::{
+    extract::{Query, State},
+    http::StatusCode,
+    response::{IntoResponse, Json},
+};
+use futures::StreamExt;
+use mongodb::{bson::doc, options::AggregateOptions};
+use serde::Deserialize;
+use starknet::core::types::FieldElement;
+use std::sync::Arc;
+
+#[derive(Deserialize)]
+pub struct StarknetIdQuery {
+    addr: FieldElement,
+}
+
+pub async fn handler(
+    State(state): State<Arc<AppState>>,
+    Query(query): Query<StarknetIdQuery>,
+) -> impl IntoResponse {
+    let domains = state
+        .starknetid_db
+        .collection::<mongodb::bson::Document>("domains");
+    let addr = to_hex(&query.addr);
+
+    let pipeline = vec![
+        doc! {
+            "$match": {
+                "legacy_address": &addr,
```

Comment:
I think this is not what you want. Instead of looking at the domains pointing towards your address you probably need to look at the domains you own (control). E.g. if I own eykar.stark and make it points towards the eykar contract, I want you to show me this domain as something I can automatically renew, same if I don't own "thisguyisaloser.stark" and someone points it towards my address, I don't want you to suggest me to renew it.